test/lifecycle_test.dart

```diff
@@ -607,6 +669,7 @@ void sharedLifecycleTests<T extends react.Component>({
       expect(
           component.lifecycleCalls,
           equals([
+            isComponent2 ? matchCall('getDerivedStateFromProps') : null,
```

Comment:
This should have `args: ` expectations to test that the correct arguments are being passed in

Reply:
Same goes for most of these; may as well add it to all of them, for consistency